### DISCUSSION TITLE
Windows installation tutorial is confusing

### DISCUSSION THREAD

**Discussion by svenja-haas** [codam-coding-college/MLX42#45] [Ideas]
hi! i just did the windows installation and a few of the bullet points are insufficient and/or confusing. I would make a few changes to the tutorial and then do a pull request if that is okay.

**Comment by W2Wizard** [comment]
Sorry for that, windows is quite a hassle indeed as there are so many steps to get the correct dependencies. But absolutely, go for it! If possible might I suggest changing it to link to the 64-Bit  version of MingW?

Thanks again!

**Reply by svenja-haas** [reply]
> Sorry for that, windows is quite a hassle indeed as there are so many steps to get the correct dependencies. But absolutely, go for it! If possible might I suggest changing it to link to the 64-Bit  version of MingW?
> 
> Thanks again!

i think the 64 bit version was the one I downloaded too, if it's mandatory to get this version i can change it yes. gonna take a while to do it since I'm busy atm, but thanks for the quick reply!

**Reply by W2Wizard** [reply]
Ok so I dabbled into it a bit, since im planning to switch to C++ as well as supporting Vulkan (along side with OpenGL) this left me in a quite a pickle to figure out how to make it easy to build projects across all platforms. This includes having the dependencies being linked super easily without taking hours to setup.

I feel like maybe switching to CMake for the building, for Unix-like systems this isn't a real problem per say, just gotta setup CMake correctly.

Windows is being quite a big pain in the ass ... However, I found the solution onto making it much much much easier than the current way.

It's basically something along the lines of:
```
- Install MSYS2
- Install these packages, run the given pacman commands:
	- pacman -S --needed base-devel mingw-w64-x86_64-toolchain
	- https://packages.msys2.org/group/mingw-w64-x86_64-vulkan-devel
	- https://packages.msys2.org/package/mingw-w64-x86_64-glfw?repo=mingw64

- In your prefered IDE set up the header include directory to be:
	- C:/msys64/mingw64/include/**

- Add C:\msys64\mingw64\bin to your environment variables in the system PATH.

- In VSCode go to the C/C++ Extension Intellisense settings page and configure with the following options:
	- CTRL + SHIFT + P -> Edit Configurations (UI)
	- Set Compiler path to: C:/msys64/mingw64/bin/g++.exe
	- Set IntelliSense mode to: linux-gcc-x64
	- Make sure for Include Path that it has these values: ${workspaceFolder}/** & C:/msys64/mingw64/include/**
	- Pick C17 & C++17 as standart!
```

Afterwards you have ALL the dependencies to run MLX42 without any header missing errors or some other vague bullshit.

It just compiles like this:
```
c++ main.cc -lglfw3 -lvulkan -lopengl32
```

Feel free to experiment and come up with a concrete setup to get it running, these were just my vague notes on how to compile on windows with MinGW64 using MSYS2.
